dipy/nn/histo_resdnn.py

```diff
@@ -0,0 +1,245 @@
+        Will not work if the declared SH_ORDER does not match the weights
+        expected input.
+
+        The weights for a sh_order of 8 can be obtained via the function:
+            get_fnames('histo_resdnn_weights').
+
+        Parameters
+        ----------
+        weights_path : str
+            Path to the file containing the weights (hdf5, saved by tensorflow)
+        """
+        try:
+            self.model.load_weights(weights_path)
+        except ValueError:
+            raise ValueError('Expected input for the provided model weights '
+                             'do not match the declared model ({})'
+                             .format(self.sh_size))
+
+    def predict(self, x_test):
+        r"""
+        Predict fODF (as SH) from input raw DWI signal (as SH)
+
+        Parameters
+        ----------
+        x_test : np.ndarray
+            Array of size (N, M) where M is
+            ``(sh_order + 1) * (sh_order + 2) / 2``.
+            N should not be too big as to limit memory usage.
+
+        Returns
+        -------
+        _ : np.ndarray (N, M)
+            Predicted fODF (as SH)
+        """
+
+        if x_test.shape[-1] != self.sh_size:
+            raise ValueError('Expected input for the provided model weights '
+                             'do not match the declared model ({})'
+                             .format(self.sh_size))
+
+        return self.model.predict(x_test)
+
+    def fit(self, data, gtab, mask=None):
+        """ Wrapper function to faciliate prediction of larger dataset.
+        The function will mask, normalize, split, predict and 're-assemble'
+        the data as a volume.
+
+        Parameters
+        ----------
+        data : np.ndarray
+            DWI signal in a 4D array
+        gtab : GradientTable class instance
+            The acquisition scheme matching the data (must contain at least
+            one b0)
+        mask : np.ndarray (optional)
+            Binary mask of the brain to avoid unnecessary computation and
+            unreliable prediction outside the brain.
+
+        Returns
+        -------
+        pred_sh_coef : np.ndarray (x, y, z, M)
+            Predicted fODF (as SH). The volume has matching shape to the input
+            data, but with ``(sh_order + 1) * (sh_order + 2) / 2`` as a last
+            dimension.
+
+        """
+        if mask is None:
+            logger.warning('Mask should be provided to accelerate '
+                           'computation, and because predictions are '
+                           'not reliable outside of the brain.')
+            mask = np.sum(data, axis=-1)
+        mask = mask.astype(bool)
+
+        # Extract B0's and obtain a mean B0
+        b0_indices = np.where(gtab.bvals < 1)[0]
+        if not len(b0_indices) > 0:
+            raise ValueError('b0 must be present for DWI normalization.')
+        logger.info('b0 indices found are: {}'.format(b0_indices))
+
+        mean_b0 = data[..., b0_indices]
```

Comment:
This is confusing because it's not mean_b0 yet. Maybe combine with line below as a one-liner? 
```suggestion
        mean_b0 = np.mean(data[..., b0_indices], axis=-1)
```